src/Execution/GraphQLRequest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Nuwave\Lighthouse\Execution;
+
+use Illuminate\Http\Request;
+use Nuwave\Lighthouse\Support\Contracts\CreatesContext;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+
+class GraphQLRequest
```

Comment:
This class basically implemented the `Iterator` interfaces for handling batches. What if we make a class for batches and make that implement `iterator` and then this class simply has an instance of the batch object?
Or make this class implement `Iterator`.

Reply:
I tried that in the first place, but found it complicates usage quite a bit. The way this is used now enables the GraphQLController to iterate through with a very simple do-while loop.

Using standardized interfaces seems good generally, maybe you can give it a shot and see if it improves things?